docs/HowTo/develop-subtopologies.md

```diff
@@ -0,0 +1,345 @@
+
+In `fprime/config`, create a file called `TopologyConfig.hpp` and include:
+
+```cpp
+#ifndef TOPOLOGYCONFIG_HPP
+#define TOPOLOGYCONFIG_HPP
+
+#include <MainDeployment/Top/MainDeploymentTopologyAc.hpp> // replace with name of your main deployment
+using namespace MainDeployment; // replace with namespace of your main deployment
+
+#endif
+```
+
+You may notice that all three functions take in the argument `state`, which is a struct that is passed in when the subtopology is called. `TopologyState` is that struct, which can include any variables you would like a developer to be able to modify. This provides dynamic-ness to your subtopology.
+
+We're now going to interject information about `MySubtopologyTopologyDefs.hpp`, because it is more relevant to the current topic.
+
+## MySubtopologyTopologyDefs.hpp
+
+In our example, we may want to allow the user to customize the clock that runs the RNG component, from 1Hz to 2Hz or otherwise. Thus, our struct may look something like:
+
+```cpp
+struct TopologyState {
+    // ...
+    U32 clockRate;
+    // ...
+}
+```
+
+This struct definition is included within `MySubtopologyDefs.hpp`, and is included into `MySubtopology.cpp` at build time. In addition to this, we may notice that we need to include global definitions in our `Def.hpp` file; however depending on your subtopology, you may find this part optional. Since we use F Prime's rate group driver, we should include "WARN" and "ERROR" flags that are tied to the status of the driver. We'll also add these to `MySubtopologyDefs.hpp`.
+
+```cpp
+namespace GlobalDefs {
+    namespace PingEntries {
+        MySubtopology_rateGroup {
+            enum { WARN = 3, ERROR = 5 };
+        }
+    } // end PingEntries
+} // GlobalDefs
+```
+
+Notice that we wrap `PingEntries` in a namespace called `GlobalDefs`. This is important, as it allows us to use the namespace `GlobalDefs` across multiple topologies, and then link them all together in the main deployment. This will become more clear in the [next section](#integration-into-a-main-deployment).
+
+This brings us to a unique naming scheme for variable names for subtopologies. On the development-end of the subtopology, we see names like `configureTopology`. However, on the build end when these subtopologies are folded into a bigger project, these functions are added to namespaces via their names. Inherently this makes sense, so that we don't confuse a function, or especially a component instance, with each other. So:
+
+```cpp
+MySubtopology -> configureTopology() ==> MySubtopology::configureTopology()
+MySubtopology -> startTopology() ==> MySubtopology::startTopology()
+MySubtopology -> rateGroup ==> MySubtopology_rateGroup // example for rateGroup instance
```

Comment:
Use a markdown table instead? Including some notes on what the names are, e.g. is `MySubtopology_rateGroup` the name of the component instance in C++ land?